Sources/MapboxDirections/Intersection.swift

```diff
@@ -156,12 +165,36 @@ extension Intersection: Codable {
         case outletRoadClasses = "classes"
         case tollCollection = "toll_collection"
         case tunnelName = "tunnelName"
+        case mapboxStreets = "mapbox_streets_v8"
         case isUrban = "is_urban"
         case restStop = "rest_stop"
         case administrativeRegionIndex = "admin_index"
         case geometryIndex = "geometry_index"
     }
     
+    /// Used to code `Intersection.outletMapboxStreetsRoadClass`
+    private struct MapboxStreetClassCodable: Codable {
+        private enum CodingKeys: String, CodingKey {
+            case streetClass = "class"
+        }
+        
+        let streetClass: MapboxStreetsRoadClass?
+        
+        init(streetClass: MapboxStreetsRoadClass?) {
+            self.streetClass = streetClass
+        }
+        
+        init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            
+            if let classString = try container.decodeIfPresent(String.self, forKey: .streetClass) {
+                streetClass = MapboxStreetsRoadClass(rawValue: classString)
+            } else {
+                streetClass = nil
```

Comment:
I think this is fine because adding new road classes is a rare event and the new value wouldn’t be immediately useful in the navigation SDK anyways. However, if we ever learn of a plan to add new road classes in Streets source v8._x_, we should preemptively implement the `other` case with an associated value, similar to what’s proposed in https://github.com/mapbox/mapbox-directions-swift/pull/506#discussion_r540281957. Not necessary just yet though.